### PR TITLE
refactor(nuxt): improve `NuxtLink` types

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

Related to https://github.com/nuxt/nuxt/issues/25532

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Exposed types from vue-router are being re-implemented, we should avoid doing this for maintenance / inconsistency reasons
- Types aren't documented, copied docs from the site
- Added auto-completion for `rel` for most common privacy, security & SEO use cases
- Improved type safety for the router link props we generate

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/components/nuxt-link.ts [hunk 1]
```diff
@@ -17,52 +24,90 @@ const firstNonUndefined = <T> (...args: (T | undefined)[]) => args.find(arg => a
...
+    to: string,
+    resolve: Router['resolve']
+  ): string
+  function resolveTrailingSlashBehavior (
+    to: RouteLocationRaw,
+    resolve: Router['resolve']
+  ): Omit<RouteLocationRaw, string>
+  function resolveTrailingSlashBehavior (
     to: RouteLocationRaw,
-    resolve: (to: RouteLocationRaw) => RouteLocation & { href?: string }
```
Comment (harlan-zw) [inline packages/nuxt/src/app/components/nuxt-link.ts:64]: Couldn't figure out why this `href` was here @manniL, maybe you know?

File: docs/3.api/1.components/4.nuxt-link.md [hunk 2]
```diff
@@ -25,6 +25,23 @@ In this example, we use `<NuxtLink>` component to link to another page of the ap
...
 
+### Handling 404s
+
+When using `<NuxtLink>` for `/public` directory files or when pointing to a different app on the same domain, you should use the `external` prop.
+
+Using `external` forces the link to be rendered as an `a` tag instead of a Vue Router `RouterLink`.
+
+```vue [pages/index.vue]
+<template>
+  <NuxtLink to="/the-important-report.pdf" external>
```
Comment (harlan-zw) [inline docs/3.api/1.components/4.nuxt-link.md:36]: It's possible that we implement this logic automatically for files, not sure if Vue Router handles file links at all?

Comment (harlan-zw) [reply docs/3.api/1.components/4.nuxt-link.md:36]: We can definitely switch it to use `a` if the `download` attribute is set which I plan to do in a separate PR

File: docs/3.api/1.components/4.nuxt-link.md [hunk 3]
```diff
@@ -25,6 +25,23 @@ In this example, we use `<NuxtLink>` component to link to another page of the ap
 </template>
 ```
 
+### Handling 404s
+
+When using `<NuxtLink>` for `/public` directory files or when pointing to a different app on the same domain, you should use the `external` prop.
```
Comment (harlan-zw) [inline docs/3.api/1.components/4.nuxt-link.md:30]: I couldn't think of any other valid use cases for manually using `external`, let me know if you think of anything

**Comment by harlan-zw** [comment]
Not too sure why but it seems like using the `RouterLinkOptions['to']` type does not provide the `typedPages` augmentation :thinking:

File: docs/3.api/1.components/4.nuxt-link.md (hunk 2 above)
Comment (posva) [reply docs/3.api/1.components/4.nuxt-link.md:36]: FYI vue-router doesn't differentiate any kind of link. They are all meant to be resolved with `router.resolve()`

Comment (harlan-zw) [reply docs/3.api/1.components/4.nuxt-link.md:36]: I see. Not sure the initial thinking of the original `NuxtLink` design but I'd like to not see this error being thrown for valid links.

> [Vue Router warn]: No match found for location with path "/the-important-report.pdf"

Guessing their may be some minor performance improvements by not using RouterLink for links that don't need them (?).

**Comment by posva** [comment]
Other than the comment I left, I don't think my review is valuable here. But I appreciate you thought it could be 😄 . Let me know if you have other vue-router questions

**Comment by harlan-zw** [comment]
> Other than the comment I left, I don't think my review is valuable here. But I appreciate you thought it could be 😄 . Let me know if you have other vue-router questions

Thanks for reviewing! Your sanity check is more than appreicated

File: packages/nuxt/src/app/components/nuxt-link.ts [hunk 4]
```diff
@@ -314,12 +362,11 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
               if (!href) { return undefined }
 
               const url = parseURL(href)
-              return {
+              return <RouteLocation & { href: string }> {
```
Comment (danielroe) [inline packages/nuxt/src/app/components/nuxt-link.ts:365]: maybe we could use `satisfies` which would ensure we actually do return values compatible with the vue router types?

File: docs/3.api/1.components/4.nuxt-link.md [hunk 5]
```diff
@@ -102,17 +130,18 @@ You can then use `<MyNuxtLink />` component as usual with your new defaults.
...
   exactActiveClass?: string;
   prefetchedClass?: string;
   trailingSlash?: 'append' | 'remove'
-}) => Component
+}
+function defineNuxtLink(options: NuxtLinkOptions): Component {}
 ```
 
-- `componentName`: A name for the defined `<NuxtLink>` component.
+- `componentName`: A name for the component. Default is `<NuxtLink>`.
```
Comment (danielroe) [inline docs/3.api/1.components/4.nuxt-link.md:144]:
```suggestion
- `componentName`: A name for the component. Default is `NuxtLink`.
```

**Review by danielroe — APPROVED** [review]

**Comment by harlan-zw** [comment]
Thanks for the merge satisfies fix 👍

**Pull request merged**